server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -167,6 +174,14 @@ FieldTypeLookup fieldTypesLookup() {
         return fieldTypeLookup;
     }
 
+    FieldTypeLookup indexTimeLookup() {
+        return indexTimeLookup;
+    }
+
+    Map<String, IndexTimeScript> indexTimeScripts() {
```

Comment:
could the index time script already have a reference to the index time lookup internally when it gets created, so that we don't need to expose this additional method in MappingLookup?

Reply:
having another look, index time lookup is needed when creating the search lookup, and I actually made another suggestion around possibly not needing IndexTimeScript entirely, so I don't think we will be able to do without adding indexTimeLooup.